src/ajax.js

```diff
@@ -254,9 +254,9 @@ function ajaxConvert( s, response, jqXHR, isSuccess ) {
 
 				// Apply converter (if not an equivalence)
 				if ( conv !== true ) {
-
-					// Unless errors are allowed to bubble, catch and return them
-					if ( conv && s[ "throws" ] ) {
+					if ( jQuery.isThenable(response) ) {
```

Comment:
I'm not a fan of introducing this abstraction. `response && jQuery.isFunction( response.then )` or `jQuery.isFunction( response && response.then )` would be better.